presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/thrift/ThriftMetastoreUtil.java

```diff
@@ -691,33 +693,40 @@ private static StorageDescriptor makeStorageDescriptor(String tableName, List<Co
     {
         TypeInfo typeInfo = columnType.getTypeInfo();
-        checkArgument(typeInfo.getCategory() == PRIMITIVE, "unsupported type: %s", columnType);
-        switch (((PrimitiveTypeInfo) typeInfo).getPrimitiveCategory()) {
-            case BOOLEAN:
-                return createBooleanStatistics(columnName, columnType, statistics);
-            case BYTE:
-            case SHORT:
-            case INT:
-            case LONG:
-                return createLongStatistics(columnName, columnType, statistics);
-            case FLOAT:
-            case DOUBLE:
-                return createDoubleStatistics(columnName, columnType, statistics);
-            case STRING:
-            case VARCHAR:
-            case CHAR:
-                return createStringStatistics(columnName, columnType, statistics, rowCount);
-            case DATE:
-                return createDateStatistics(columnName, columnType, statistics);
-            case TIMESTAMP:
-                return createLongStatistics(columnName, columnType, statistics);
-            case BINARY:
-                return createBinaryStatistics(columnName, columnType, statistics, rowCount);
-            case DECIMAL:
-                return createDecimalStatistics(columnName, columnType, statistics);
-            default:
-                throw new IllegalArgumentException(format("unsupported type: %s", columnType));
+        if (typeInfo.getCategory() == PRIMITIVE) {
+            switch (((PrimitiveTypeInfo) typeInfo).getPrimitiveCategory()) {
+                case BOOLEAN:
+                    return createBooleanStatistics(columnName, columnType, statistics);
+                case BYTE:
+                case SHORT:
+                case INT:
+                case LONG:
+                    return createLongStatistics(columnName, columnType, statistics);
+                case FLOAT:
+                case DOUBLE:
+                    return createDoubleStatistics(columnName, columnType, statistics);
+                case STRING:
+                case VARCHAR:
+                case CHAR:
+                    return createStringStatistics(columnName, columnType, statistics, rowCount);
+                case DATE:
+                    return createDateStatistics(columnName, columnType, statistics);
+                case TIMESTAMP:
+                    return createLongStatistics(columnName, columnType, statistics);
+                case BINARY:
+                    return createBinaryStatistics(columnName, columnType, statistics, rowCount);
+                case DECIMAL:
+                    return createDecimalStatistics(columnName, columnType, statistics);
+                default:
+                    throw new IllegalArgumentException(format("unsupported type: %s", columnType));
+            }
         }
+
+        if (typeInfo.getCategory() == STRUCT || typeInfo.getCategory() == MAP) {
```

Comment:
@sujay-jain LIST type needs to be handled as well, right? Otherwise ANALYZE will fail on tables with columns of array/list type.

Reply:
maybe just make this an `else` to make sure we aren't missing anything else. I don't see a reason we need to throw an unsupported type exception for anything if we can return null stats.